[RC2 CH03] Add introduction (#1832)
cntt-n/CNTT: doc/ref_cert/RC2/chapters/chapter03.md

# 3. Kubernetes Test Cases and Traceability to CNTT Requirements
<p align="right"><img src="../figures/bogo_ifo.png" alt="scope" title="Scope" width="35%"/></p>

## Introduction

> All of the requirements for RC2 have been defined in the Reference Model (RM) and Reference Architecture (RA). The scope of this chapter is to identify and list down test cases based on these requirements. Users of this chapter will be able to use it to determine which test cases they must run in order to test compliance with requirements. This will enable traceability between test cases and requirements. They should be able to clearly see which requirements are covered by which tests and the mapping from a specific test result (pass or fail) to the requirement. Each requirement may have one or more test case associated with it.

Reviewer: Could a test case cover more than one requirement? If yes, then would need to add sentence for that. 
Minor typos:
```suggestion
All of the requirements for RC2 have been defined in the Reference Model (RM) and Reference Architecture (RA2). The scope of this chapter is to identify and list down test cases based on these requirements. Users of this chapter will be able to use it to determine which test cases they must run in order to test compliance with the requirements. This will enable traceability between the test cases and requirements. They should be able to clearly see which requirements are covered by which tests and the mapping from a specific test result (pass or fail) to a requirement. Each requirement may have one or more test case associated with it.
```

### Goals

> - Clear mapping between requirements and test results

Reviewer: is it "test results" or "test cases"?

> - Clear mapping between requirements and test results

Author: ```suggestion
- Clear mapping between requirements and test cases
```

> - Clear mapping between requirements and test results

Author: ```suggestion
- Clear mapping between requirements and test cases
```

- Provide a stable set of point-in-time requirements and tests to achieve conformance

> - Enable clear tracability with versioning to know which requirements have and haven't been covered and track changes over time

Reviewer 2: I assume this refers to the various versions of the RC document, i.e., the traceability is enabled by comparing different versions of RC documents. This statement does not imply that the RC document itself contains such historic data. This would also contradict the previous point a bit.

Author: I was reading this as each RC document has traceability to one set of point in time requirements while the RC document can evolve from version to version. Is there a way we could write this more clearly?

Reviewer 2: How about something along the lines of
```suggestion
- Enable clear traceability of the coverage of requirements across consecutive releases of this document
```

- Clickable links from test cases to requirements

> - One or more tests for every MUST requirement

Reviewer 2: Another explicit goal could be: The list of test cases serves as input/blueprint/template/guidance for OVP (2.0).

Author: added

> - One or more tests for every MUST requirement

Author: ```suggestion
- One or more tests for every MUST requirement
- A set of test cases to serve as a template for Cloud Native OVP
```

### Non-Goals
- Defining any requirements
- Providing coverage for non-testable requirements

### Definitions
*must*: Test Cases that are marked as must are considered mandatory and must pass successfully

> *should*: Test Cases that are marked as should are expected to be fulfilled by the cloud infrastructure but it is up to each service provider whether to accept a cloud infrastructure that is fulfilling any of these requirements. The same applies to should not.

Author: ```suggestion
*should*: Test Cases that are marked as should are expected to be fulfilled by the cloud infrastructure but it is up to each service provider whether to accept a cloud infrastructure that is not fulfilling any of these requirements. The same applies to should not.
```

> *should*: Test Cases that are marked as should are expected to be fulfilled by the cloud infrastructure but it is up to each service provider whether to accept a cloud infrastructure that is not reflecting on any of those requirements. The same applies to should not.

Reviewer: ```suggestion
*should*: Test Cases that are marked as should are expected to be fulfilled by the cloud infrastructure but it is up to each service provider whether to accept a cloud infrastructure that is fulfilling any of these requirements. The same applies to should not.
```
